api/src/main/java/org/openmrs/Condition.java

```diff
@@ -255,7 +255,7 @@ public Date getOnsetDate() {
 	 * @param onsetDate the onset date of the condition to be set
 	 */
 	public void setOnsetDate(Date onsetDate) {
-		this.onsetDate = onsetDate;
+		this.onsetDate = new Date(onsetDate.getTime());
```

Comment:
What happens, if the `onsetDate` is null?

Reply:
Well I have fixed that issue, we can simply set those dates to null and cause NullPointerException in case user passes null values to prevent them from doing so